smalltalksrc/VMMaker/SpurMemoryManager.class.st

```diff
@@ -10231,6 +10288,20 @@ SpurMemoryManager >> outerContextOf: aBlockClosureOop [
 	^ self fetchPointer: FullClosureOuterContextIndex ofObject: aBlockClosureOop
 ]
 
+{ #category : 'interpreter access' }
+SpurMemoryManager >> paddingForFormatInstBytes: instBytes forElementBytes: elementSizeInBytes shiftForElement: shiftForElement [
+
+	"This method is doing the same as below but using bit arithmetic.
+	shiftForElement SHOULD be log 2 of element size in bytes. We don't do the log2 because 1) is slow 2) it can return floats
+	
+	slotsPerWord := (wordSize / elementSize).
+	trailingBytes := instBytes \\ wordSize.
+	trailingSlots := trailingBytes / elementSize.
+	^ slotsPerWord - trailingSlots. "
+
+	^ self wordSize // elementSizeInBytes - (instBytes >> shiftForElement) bitAnd: self wordSize // elementSizeInBytes - 1
```

Comment:
There's a bug here when `self wordSize < elementSizeInBytes` (in those cases we never have padding).

The ugly fix is to add an if returning zero on that case.